ui: Use structure-icons as much as possible

johncowen opened the pull request:
This PR removes as many as our old icons as possible, in favour of `structure-icons`, and is partly as a result of working to far on icons during (https://github.com/hashicorp/consul/pull/6639).

Once both https://github.com/hashicorp/consul/pull/6639 and this PR is merged down, we should be 100% using our `base` CSS - which is almost 100% `structure-icons`. Outstanding icons that we have in `base` but not in `structure-icons` we'll work to get them integrated.

Following merge of this and https://github.com/hashicorp/consul/pull/6639 we should be able to get rid of our project `components/icons/index.scss` file 🎉

opihana commented:
I noticed the inputs on multiple pages are janky now:

![Screen Shot 2019-12-02 at 3 03 52 PM](https://user-images.githubusercontent.com/6323933/69995087-0a541300-1515-11ea-9183-85135b3a21c0.png)

![Screen Shot 2019-12-02 at 3 03 19 PM](https://user-images.githubusercontent.com/6323933/69995088-0a541300-1515-11ea-83cf-4da5af4e44e8.png)

And the alignment of a few icons in titles and subtitles is inconsistent in its vertical alignment.

![Screen Shot 2019-12-02 at 3 03 05 PM](https://user-images.githubusercontent.com/6323933/69995089-0a541300-1515-11ea-8394-fc92ea57526b.png)

![Screen Shot 2019-12-02 at 3 05 16 PM](https://user-images.githubusercontent.com/6323933/69995152-2bb4ff00-1515-11ea-845c-bfe753615a53.png)

johncowen commented:
Hey @opihana !

Ok I rebased https://github.com/hashicorp/consul/pull/6843 onto here which was the fix for the first 2 screengrabs.

The other two screengrabs were not changed at all in this PR and have been like that for a while:

- External Service Icons.
 I remember originally wondering how this should be lined up, eventually I went with the bottom on the baseline as this also happens to line the top of the icon up with the top of a lowercase letter and lines up 'ok' with the proxy label:

![Screenshot 2019-12-04 at 12 28 06](https://user-images.githubusercontent.com/554604/70143411-72306800-1693-11ea-8a36-47cab9fc27f4.png)

I can't find these icons in the Sketch designs anymore, but maybe we should get those back in there (or could you point me to them if they are somewhere I've not seen) and I can follow whatever you do there. We should definitely decide how this entire line should be vertically positioned (P.S. I found an old design with these in, but the design doesn't have the proxy label in which is why I think this troublesome when I first did it)

- Info icons. I think I might have changed these a while ago (so not in this PR), and the (i) in that screengrab does actually look bigger than it should be. But it has always been in a 'superscript' position. 

Actually looking at it more closely, there seems to be a Chrome bug in play here. If you resize the browser window the (i) icon occasionally jumps to be a pixel or so larger than it should be even though no CSS changes. I checked in FF and Safari but it doesn't happen there. Looks to be a little browser bug, but I'd probably look at that in a separate PR if at all.

johncowen commented:
This is currently on hold pending a rebase of `ui-staging` and then re-checking.

johncowen commented:
This has been rebased and is ready again for review. We also added one last commit (now that we have rebased from other feature work) to remove the remaining CSS in our app level `icons/index` file. 100% of our icons are now provided by `base`. The only thing that is icon related is our `%external-source-icon` helper, which we've moved to `%app-view` for the moment pending a rename.

`base` still contains some extra svg icons that are either using our old way of colorizing icons (just duplicating the svg and hardcoding the color), or icons that don't yet exist in `@hashicorp/structure-icons`. We'll be working to move all our old style icon coloring to move everything to use masking in a future PR and also working to try to get icons that don't exist in `@hashicorp/structure-icons` integrated into that module.

opihana commented:
@johncowen Does the preview link in the original comment stay valid? If so, I noticed that there are a few issues:

There's no content on the Settings page

<img width="1438" alt="Screen Shot 2019-12-18 at 4 30 14 PM" src="https://user-images.githubusercontent.com/6323933/71128872-daaf3700-21b3-11ea-8b57-84de521a8622.png">

And the intentions were doing this weird swapping of the name when I click on it. Watch carefully! 

![intentions-swap](https://user-images.githubusercontent.com/6323933/71128891-e7cc2600-21b3-11ea-8f0f-1cb949a82c05.gif)

johncowen commented:
> There's no content on the Settings page

Rebased https://github.com/hashicorp/consul/pull/6965

> And the intentions were doing this weird swapping of the name when I click on it. Watch carefully!

This is a mix of our mocks and ember:

1. Are mocks are partly random and there is no state. So when you click on an intention with the id of 05b1c5f2-4729-5647-122d-7042372ca72f on the intention listing, when you arrive on the edit page you'll notice that the values to edit are different. This is because the mocks are stateless and everything is generated 'randomly but consistent' (using a salt).
2. Despite point 1. Ideally you still shouldn't see this little flicker as ember/ember-data/the framework should teardown the listing view before it loads any data for the next view, therefore you wouldn't see the little flicker/change when you click the item. What causes the flicker is that the different values for the intention that you clicked on are loaded in before it gets chance to teardown the list view and setup the edit view. This is something I've seen before in other ember apps.

As point 1 is unlikely to happen in production you are unlikely to see this in reality (there is a chance that the intention could be edited by other means before you've clicked on it, therefore you would see this, but if this does happen this problem is has been in the UI since the start and I'd guess we are unlikely to look into it - one suggestion would be that we use something other than ember-data to see if that resolves the issue. I'm not sure if this exact thing is inherent with ember or ember-data.

All in all this PR only has CSS changes (and one tiny HTML change) just to replace icons, so anything javascript-y like this won't have been added via this PR.

johncowen commented:
@opihana are you ok to re-review this, or would you rather I add someone else to review this?

johncowen commented on ui-v2/app/styles/components/app-view/skin.scss:
```diff
@@ -1,3 +1,10 @@
+/*TODO: Rename this to %app-view-brand-icon or similar */
+%with-external-source-icon {
```
If I follow what you are saying here, that would be a great idea for if there were other places that we don't own that were using these CSS files:

```
%app-view-brand-icon {
  @extend %with-external-source-icon;
}
```
then both placeholders would do the same thing, and we could gradually deprecate the old one.

As we own the entire source here it probably just as good to search/replace the instances where we use `%external-source-icon` with the new name, might be something you could PR if you fancied it?

All that ^ is if I've understood correctly - we can go over it later.
